### PR TITLE
Jaxb context citavi xml importer 9539

### LINKED ISSUE #9539: Rewrite code of CitaviImporter to avoid JAXBContext
The code of [`org.jabref.logic.importer.fileformat.CitaviXmlImporter`](https://github.com/JabRef/jabref/blob/2db30cb26d1140af307042126fe2a3f0ba121f06/src/main/java/org/jabref/logic/importer/fileformat/CitaviXmlImporter.java#L61) is written OK, but relies on JAXB.  It should be rewritten using [StAX-Parser](https://mkyong.com/java/how-to-read-xml-file-in-java-stax-parser/) and thus getting rid of a JAXB dependency.

The code reads like we do not hit polymorphism in XMLElements (which is not supported by JAXB. See https://github.com/FasterXML/jackson-modules-base/issues/127).

See https://github.com/JabRef/jabref/blob/d21c11337a17af85047d763218c9df4dbc4db1be/src/main/java/org/jabref/logic/importer/fileformat/MedlineImporter.java#L114 for some code hints.

Similar to https://github.com/JabRef/jabref/pull/9673, the gradle build target `generateCitaviSource` can be removed and the `xsd` file be removed from the source repository.

### Example file

1. Open 7Zip gui
2. Open `src/test/resources/org/jabref/logic/importer/fileformat/CitaviXmlImporterTest4.ctv6bak` in 7zip.
3. Open `citavi-single.xml` using 7 zip

### Initial coding

1. Remove gradle task `generateCitaviSource`
4. Remove `src/main/resources/xjc/citavi/citavi.xsd`
5. Exectue gradle `clean` task.
6. Observe that `CitaviXmlImporter` does not compile
7. Try to understand how MedlineImporter handles XML importing
8. Work on fixing it. -> You have a) an example file (see above) and b) similar code (MedlineImporter )

## Negleted Implementation Option

Thus, it seems to be "just" to include the dependency to [jackson module jakarta-xmlbind](https://github.com/FasterXML/jackson-modules-base/tree/2.15/jakarta-xmlbind) and to use Jackson's XMLMapper. Code examples are at https://stackify.com/java-xml-jackson/.

### PR DESCRIPTION
**Pull request by psyren99**
### Mandatory checks

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -126,8 +126,6 @@ repositories {
     maven { url 'https://s01.oss.sonatype.org/content/repositories/snapshots/' }
     maven { url 'https://jitpack.io' }
     maven { url 'https://oss.sonatype.org/content/groups/public' }
-
-    // Required for one.jpro.jproutils:tree-showing
```
Comment (koppor) [inline build.gradle:130]: Do not remove comments

File: build.gradle [hunk 2]
```diff
@@ -199,7 +197,11 @@ dependencies {
     implementation group: 'org.eclipse.jgit', name: 'org.eclipse.jgit', version: '6.10.0.202406032230-r'
 
     implementation group: 'com.fasterxml.jackson.dataformat', name: 'jackson-dataformat-yaml', version: '2.17.2'
+    implementation group: 'com.fasterxml.jackson.dataformat', name: 'jackson-dataformat-xml', version: '2.17.2'
     implementation group: 'com.fasterxml.jackson.datatype', name: 'jackson-datatype-jsr310', version: '2.17.2'
+    implementation group: 'com.fasterxml.jackson.module', name: 'jackson-module-jaxb-annotations', version: '2.17.2'
+    implementation group: 'com.fasterxml.jackson.module', name: 'jackson-module-jakarta-xmlbind-annotations', version: '2.13.0'
```
Comment (koppor) [inline build.gradle:203]: Are these really needed? I thought, we use StaX

File: build.gradle [hunk 3]
```diff
@@ -199,7 +197,11 @@ dependencies {
...
+
```
Comment (koppor) [inline build.gradle:204]: Remove empty line

File: src/main/java/module-info.java [hunk 4]
```diff
@@ -81,7 +81,9 @@
     requires com.google.gson;
     requires com.fasterxml.jackson.databind;
     requires com.fasterxml.jackson.dataformat.yaml;
+    requires com.fasterxml.jackson.dataformat.xml;
     requires com.fasterxml.jackson.datatype.jsr310;
+    requires com.fasterxml.jackson.module.jaxb;
```
Comment (koppor) [inline src/main/java/module-info.java:86]: REally required? I thought, we use Stax.

**Review by koppor — CHANGES_REQUESTED** [review]
Just a quick review with the obvious catches.

Additnoally, are these files really needed - or are these just results of adding extracted files?

image

**Comment by koppor** [comment]
> should i open a new branch? seems to still have an issue.

GitHub shows an OKish diff.

However, the tests fail. Can you please have a look?

image

**Comment by koppor** [comment]
something is wrong with file recognition. Is an unzip missing?

image

File: src/main/java/org/jabref/logic/importer/fileformat/CitaviXmlImporter.java [hunk 5]
```diff
@@ -458,14 +367,32 @@ private BufferedReader getReaderFromZip(Path filePath) throws IOException {
...
+
         // Citavi XML files sometimes contains BOM markers. We just discard them.
         // Solution inspired by https://stackoverflow.com/a/37445972/873282
-        return new BufferedReader(
-                new InputStreamReader(
-                        new BOMInputStream(
-                                Files.newInputStream(newFile, StandardOpenOption.READ),
-                                false,
-                                ByteOrderMark.UTF_8, ByteOrderMark.UTF_16BE, ByteOrderMark.UTF_16LE, ByteOrderMark.UTF_32BE, ByteOrderMark.UTF_32LE)));
+        return new BufferedInputStream(new BOMInputStream(Files.newInputStream(newFile, StandardOpenOption.READ), false, ByteOrderMark.UTF_8, ByteOrderMark.UTF_16BE, ByteOrderMark.UTF_16LE, ByteOrderMark.UTF_32BE, ByteOrderMark.UTF_32LE));
```
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/fileformat/CitaviXmlImporter.java:395]: Why the reformat?

File: src/main/java/org/jabref/logic/importer/fileformat/CitaviXmlImporter.java [hunk 6]
```diff
@@ -458,14 +367,32 @@ private BufferedReader getReaderFromZip(Path filePath) throws IOException {
...
+    }
+
+    private static InputStream checkForUtf8BOMAndDiscardIfAny(InputStream inputStream) throws IOException {
+        PushbackInputStream pushbackInputStream = new PushbackInputStream(new BufferedInputStream(inputStream), 3);
+        byte[] bom = new byte[3];
+        if (pushbackInputStream.read(bom) != -1) {
+            if (!((bom[0] == (byte) 0xEF) && (bom[1] == (byte) 0xBB) && (bom[2] == (byte) 0xBF))) {
+                pushbackInputStream.unread(bom);
+            }
+        }
```
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/fileformat/CitaviXmlImporter.java:388]: Why the same functionality twice?

File: src/main/java/org/jabref/logic/importer/fileformat/CitaviXmlImporter.java [hunk 7]
```diff
@@ -458,14 +367,32 @@ private BufferedReader getReaderFromZip(Path filePath) throws IOException {
...
+
+        Path newFile = Files.createTempFile("citavicontent", ".xml");
+        newFile.toFile().deleteOnExit();
```
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/fileformat/CitaviXmlImporter.java:391]: Why a `newFile` is generated here?

**Review by koppor — CHANGES_REQUESTED** [review]
Small comment regarding your reading code. I think, this is also the answer to the question why the tests are failing.

Please try to debug `org.jabref.logic.importer.fileformat.CitaviXmlImporterFilesTest#isRecognizedFormat`. Meaning: step through and think what the machine is doing.

---

I am curious: How much AI-support of which service did you use for this PR?

File: build.gradle [hunk 8]
```diff
@@ -199,7 +199,10 @@ dependencies {
     implementation group: 'org.eclipse.jgit', name: 'org.eclipse.jgit', version: '6.10.0.202406032230-r'
 
     implementation group: 'com.fasterxml.jackson.dataformat', name: 'jackson-dataformat-yaml', version: '2.17.2'
+    implementation group: 'com.fasterxml.jackson.dataformat', name: 'jackson-dataformat-xml', version: '2.17.2'
     implementation group: 'com.fasterxml.jackson.datatype', name: 'jackson-datatype-jsr310', version: '2.17.2'
+    implementation group: 'com.fasterxml.jackson.module', name: 'jackson-module-jaxb-annotations', version: '2.17.2'
+    implementation group: 'javax.xml.bind', name: 'jaxb-api', version: '2.3.1'
```
Comment (koppor) [inline build.gradle:205]: I already told that this is not necessary! Proof me wrong otherwise.

**Comment by koppor** [comment]
This refs #9539

File: README.md [hunk 9]
```diff
@@ -89,7 +89,7 @@ We will discuss improvements with you and agree to merge them once the [develope
 
 If you want a step-by-step walk-through on how to set-up your workspace, please check [this guideline](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace).
 
-To compile JabRef from source, you need a Java Development Kit 21 and `JAVA_HOME` pointing to this JDK.
+To compile JabRef from source, you need a Java Development Kit 20 and `JAVA_HOME` pointing to this JDK.
```
Comment (Siedlerchr) [inline README.md:92]: why do you change this?

Comment (koppor) [reply README.md:92]: That should be reverted. -- No changes any more. 

Based on this proposal, I submitted https://github.com/JabRef/jabref/pull/11660

**Comment by koppor** [comment]
Converted to draft, since many tests are red.

**Review by koppor — CHANGES_REQUESTED** [review]
Please fix checks and tests

**Comment by koppor** [comment]
Closing this issue due to inactivity :zzz:  Please ping us if you intend to resume work on this one.

**Pull request closed without merging**